### DISCUSSION TITLE
Calling isset on computed property always returns false

### DISCUSSION THREAD

**Discussion by smares** [livewire/livewire#3346] [1. Help] [answered]
If I have something like

```
public function getFooProperty(): ?int
{
    return 1;
}
```

and in the Blade file

```
@isset($this->foo)
```

is it normal that the block is always skipped?

`@dd(isset($this->foo))` prints `false` while `@dd($this->foo)` prints `1`.

Any chance only `__get()` was implemented and `__isset()` not?

**Comment by Dimonka2** [comment]
`@isset()` directive (or PHP function) is not really applicable to objects and especially to virtual properties. You can create your own blade directive for checking `null` values of object properties, like `isnull()`. 
So it seems for me that the topic is unrelated to Livewire.

**Reply by smares** [reply]
Why shouldn't it be? It's the same as with the `__get` magic method.

**Reply by Dimonka2** [reply]
Because documentation of `isset` [says ](https://www.php.net/manual/en/function.isset.php)
> isset — Determine if a variable is declared and is different than null

And if you scroll down comments there, you will find that `isset` does not work on virtual properties by design.

[`property_exists`](https://www.php.net/manual/en/function.property-exists.php) will also fail on virtual properties 

Thus you need to check `if(!is_null($instance->property))`. Or explicitly pass property to blade using `view('myView')->with('myProperty', $this->myProperty)`

**Comment by smares** [comment, marked as answer]
This was eventually implemented with PR https://github.com/livewire/livewire/pull/4375
